index.rst

```diff
@@ -1,61 +1,365 @@
+    Frozen accounts still hold the account name and do not allow it to be reused.
+
+IDM-1002
+    Administrators of the Science Platform must be able to delete accounts.
+    This is normally used for mistakenly-created accounts, not for accounts that were legitimate and active but should no longer be allowed access.
+
+IDM-1003
+    It must be possible to set an expiration date on an account.
+    This can be done by Science Platform administrators, or by the person approving access in the IDM-0003 use case.
+    When the expiration date arrives, the account must be automatically frozen.
+
+IDM-1004
+    Users must be notified via email of upcoming account expiration so that they can investigate renewal options if needed.
+
+Metadata
+--------
+
+IDM-1100
+    A full name must be associated with each account and prepopulated with information from the identity provider.
+    The user must be able to change the full name to anything they wish.
+
+IDM-1101
+    An email address must be associated with each account and prepopulated with information from the identity provider.
+    The user must be able to change the email address to anything they wish, but they must then verify that the email address is valid and owned by them by responding to a challenge sent to that email address.
+    Challenges for an email address must not contain user-provided content so that they cannot be used for spamming purposes.
+
+Quotas
+------
+
+IDM-1200
+    Users may have one or more quota grants associated directly with their account.
+    These may represent file storage quotas or any other service limit that may vary by user (API rate limits, CPU equivalents for batch jobs, download size limits, or whatever may eventually be appropriate).
+    The identity management system need not understand the quotas, but it should be able to sum multiple quotas under the same label.
+
+IDM-1201
+    The user must be able to view all of their existing quotas.
+
+IDM-1202
+    The user must be able to request a new quota grant.
+    That request should be routed to some approval process by a manager of the relevant resource, who can then grant or deny the request via the identity management web interface.
+
+IDM-1203
+    Quota grants may expire.
+    The user must be notified via email of pending quota grant expirations.
+
+Administration
+--------------
+
+IDM-1300
+    Administrators of the Science Platform must be able to modify any of the user's metadata on behalf of the user.
+
+IDM-1301
+    Administrators of the Science Platform must be able to set and change expiration dates on accounts.
+
+IDM-1302
+    Administrators of the Science Platform must be able to approve a pending change of email address even if the user has not responded to the challenge.
+
+IDM-1303
+    Administrators of the Science Platform must be able to create, revoke, and change the expiration dates on quota grants.
+
```

Comment:
Administrators should be able to revoke user tokens.  This could be done via the impersonation capability, but it seems more natural to have a direct interface for it; this limits impersonation actions only to those cases where it's truly necessary.